include/openthread/dataset.h

```diff
@@ -515,6 +517,21 @@ otError otDatasetGeneratePskc(const char *           aPassPhrase,
                               const otExtendedPanId *aExtPanId,
                               otPskc *               aPskc);
 
+/**
+ * This function sets an `otNetworkName` instance from a given null terminated C string.
+ *
+ * This function also validates that the given @p aNameString follows UTF-8 encoding and its length is not longer than
+ * `OT_NETWORK_NAME_MAX_SIZE`.
+ *
+ * @param[out] aNetworkName        A pointer to the `otNetworkName` to set.
+ * @param[in]  aNameString         A name C string.
+ *
+ * @retval OT_ERROR_NONE           Successfully set @p aNetworkName from @p aNameString.
+ * @retval OT_ERROR_INVALID_ARGS   @p aNameStrng is invalid (too long or does not follow UTF-8 encoding).
+ *
+ */
+otError otNetworkNameSet(otNetworkName *aNetworkName, const char *aNameString);
```

Comment:
We have other similar APIs that use `FromString()` rather than `Set()`. Thoughts?
```suggestion
otError otNetworkNameFromString(otNetworkName *aNetworkName, const char *aNameString);
```

Reply:
Thanks for suggestion. I also like `FromString()` name better. Changed it in latest push.